pkg/skaffold/docker/dependencies.go

```diff
@@ -42,30 +54,48 @@ func NormalizeDockerfilePath(context, dockerfile string) (string, error) {
 	return filepath.Abs(rel)
 }
 
-// GetDependencies finds the sources dependencies for the given docker artifact.
+// GetDependencies finds the sources dependency for the given docker artifact.
 // All paths are relative to the workspace.
 func GetDependencies(ctx context.Context, workspace string, dockerfilePath string, buildArgs map[string]*string, cfg Config) ([]string, error) {
 	absDockerfilePath, err := NormalizeDockerfilePath(workspace, dockerfilePath)
 	if err != nil {
 		return nil, fmt.Errorf("normalizing dockerfile path: %w", err)
 	}
 
-	// If the Dockerfile doesn't exist, we can't compute the dependencies.
+	deps, _ := sfGetDependencies.Do(absDockerfilePath, func() (interface{}, error) {
+		if dep, ok := dependencyCache.Load(absDockerfilePath); ok {
```

Comment:
Since `dependencyCache` is only accessed within the `singleflight` object it doesn't need concurrency safe type. You should be able to use a regular `map[string]interface{}` instead of `sync.Map`.

Reply:
I think I'm wrong. I didn't think about that different keys would cause multiple goroutines.

Reply:
correct. since multiple go routines can come in and read/write the map for different keys.